### PR TITLE
docker: support the client only in darwin

### PR DESCRIPTION
**Pull request by aespinosa**
Can probably work in other platforms but I only tested this on OSX 10.9.5 

cc @offlinehacker @tailhook

### CODE REVIEW

**Comment by fazo96** [comment]
maybe should be implemented as a `clientOnly` argument that gets overridden to `true` on darwin and  possibly other environments when the server wouldn't work?

**Comment by offlinehacker** [comment]
Looks good to me

**Comment by aespinosa** [comment]
Added the clientOnly flag @fazo96

File: pkgs/applications/virtualization/docker/default.nix [hunk 1]
```diff
@@ -48,6 +57,6 @@ stdenv.mkDerivation rec {
     description = "An open source project to pack, ship and run any application as a lightweight container";
     license = licenses.asl20;
     maintainers = with maintainers; [ offline tailhook ];
-    platforms = platforms.linux;
```
Comment (offlinehacker) [inline pkgs/applications/virtualization/docker/default.nix]: You can use 'optional clientOnly platform.darwin' here instead, just as a note, don't need to change now.

File: pkgs/top-level/all-packages.nix [hunk 2]
```diff
@@ -10954,7 +10954,14 @@ let
   dmtx-utils = callPackage (callPackage ../tools/graphics/dmtx-utils) {
   };
 
-  docker = callPackage ../applications/virtualization/docker { go = go_1_4; };
+  docker = callPackage ../applications/virtualization/docker {
+    go = go_1_4;
+    clientOnly = 
+      if stdenv.isDarwin then
```
Comment (offlinehacker) [inline pkgs/top-level/all-packages.nix]: No need for if here, IsDarwin is boolean already.

File: pkgs/applications/virtualization/docker/default.nix [hunk 3]
```diff
@@ -14,7 +14,11 @@ stdenv.mkDerivation rec {
     sha256 = "0nwd5wsw9f50jh4s5c5sfd6hnyh3g2kmxcrid36y1phabh30yrcz";
   };
 
-  buildInputs = [ makeWrapper go sqlite lxc iproute bridge-utils devicemapper btrfsProgs iptables e2fsprogs ];
+  buildInputs = [ makeWrapper go sqlite ] ++
+    (if clientOnly then
```
Comment (offlinehacker) [inline pkgs/applications/virtualization/docker/default.nix]: Use optinals here instead.

File: pkgs/applications/virtualization/docker/default.nix (hunk 1 above)
Comment (aespinosa) [inline pkgs/applications/virtualization/docker/default.nix]: done

**Comment by aespinosa** [comment]
thanks for the feedback @offlinehacker . I wasn't familiar with stdenv.lib.optional[s] before. I hope this works now.

**Comment by aespinosa** [comment]
@offlinehacker @fazo96 , this pull request is ready for review again.

**Pull request closed without merging**